src/Core/AccurateComparison.h

```diff
@@ -515,15 +515,24 @@ inline bool NO_SANITIZE_UNDEFINED convertNumeric(From value, To & result)
         return true;
     }
 
-    /// Note that NaNs doesn't compare equal to anything, but they are still in range of any Float type.
-    if (isNaN(value) && std::is_floating_point_v<To>)
+    if constexpr (std::is_floating_point_v<From> && std::is_floating_point_v<To>)
     {
-        result = value;
-        return true;
+        /// Note that NaNs doesn't compare equal to anything, but they are still in range of any Float type.
+        if (isNaN(value))
+        {
+            result = value;
+            return true;
+        }
+    }
+
+    if (accurate::greaterOp(value, std::numeric_limits<To>::max())
+        || accurate::greaterOp(std::numeric_limits<To>::min(), value))
+    {
+        return false;
     }
 
     result = static_cast<To>(value);
-    return equalsOp(value, result);
+    return true;
```

Comment:
Also what's about the case of conersion 1.5 to integer type (should return false)?

PS. Exact equality is Ok, so 1.000...1 also not equals to 1.

Reply:
Agreed equalsOp is required.